src/neynarAPI/NeynarAPIClient.ts

```diff
@@ -0,0 +1,679 @@
+    if (typeof castOrCastHash === "string") {
+      castHash = castOrCastHash;
+    } else {
+      castHash = castOrCastHash.hash;
+    }
+    try {
+      const response = await this.apis.v2Cast.cast(
+        CastParamType.Hash,
+        castHash
+      );
+      return response.data.cast ?? undefined;
+    } catch (error) {
+      if (NeynarAPIClient.isApiErrorResponse(error)) {
+        const status = error.response.status;
+        if (status === 404) {
+          return undefined;
+        }
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * Publishes a cast for the currently authenticated user
+   */
+  public async publishCast(
+    signerUuid: string,
+    text: string,
+    replyTo?: v2Cast | { hash: string },
+    embeds?: CastEmbeds[]
+  ): Promise<PostCastResponseCast> {
+    const body: PostCastReqBody = {
+      signer_uuid: signerUuid,
+      text: text,
+      embeds: embeds,
+      parent: replyTo !== undefined ? replyTo.hash : undefined,
+    };
+    const response = await this.apis.v2Cast.postCast(body);
+    return response.data.cast;
+  }
+
+  /**
+   * Delete a cast
+   */
+  public async deleteCast(
+    signerUuid: string,
+    castOrCastHash: v2Cast | string
+  ): Promise<void> {
+    let castHash: string;
+    if (typeof castOrCastHash === "string") {
+      castHash = castOrCastHash;
+    } else {
+      castHash = castOrCastHash.hash;
+    }
+    const body: DeleteCastReqBody = {
+      signer_uuid: signerUuid,
+      target_hash: castHash,
+    };
+    await this.apis.v2Cast.deleteCast(body);
+  }
+
+  /**
+   * React to a cast
+   */
+  public async reactToCast(
+    signerUuid: string,
+    reaction: ReactionType,
+    castOrCastHash: v2Cast | string
+  ): Promise<OperationResponse> {
+    let castHash: string;
+    if (typeof castOrCastHash === "string") {
+      castHash = castOrCastHash;
+    } else {
+      castHash = castOrCastHash.hash;
+    }
+    const body: ReactionReqBody = {
+      signer_uuid: signerUuid,
+      reaction_type: reaction,
+      target: castHash,
+    };
+    const response = await this.apis.reaction.postReaction(body);
+    return response.data;
+  }
+
+  /**
+   * Remove a reaction to a cast
+   */
+  public async removeReactionToCast(
+    signerUuid: string,
+    reaction: ReactionType,
+    castOrCastHash: v2Cast | string
+  ): Promise<OperationResponse> {
+    let castHash: string;
+    if (typeof castOrCastHash === "string") {
+      castHash = castOrCastHash;
+    } else {
+      castHash = castOrCastHash.hash;
+    }
+    const body: ReactionReqBody = {
+      signer_uuid: signerUuid,
+      reaction_type: reaction,
+      target: castHash,
+    };
+    const response = await this.apis.reaction.deleteReaction(body);
+    return response.data;
+  }
+
+  /**
+   * Fetches casts in a given thread.
+   * Note that the parent provided by the caller is included in the response.
+   */
+  public async fetchCastsInThread(
+    threadParent: v1Cast | { hash: string }
```

Comment:
How's this work for threads around channels or URLs?

Reply:
it only works for casts unfortunately - https://docs.neynar.com/reference/get-all-casts-in-thread

Reply:
it only works for casts unfortunately - https://docs.neynar.com/reference/get-all-casts-in-thread